Create plasmapy.analysis sub-pkg & add fit functions as initial functionality (#908)
PlasmaPy/PlasmaPy: docs/ad/index.rst

.. include:: ad_attention.inc

.. _ad:

==============================
Analysis & Diagnostic Toolkits
==============================

> Analyses and diagnostics go hand-in-hand, but have subtle differences.  Thus,

Reviewer: Eventually we'll want to have analyses related to simulations, for example, so we'll eventually want to change the wording a bit.  But, this wording is fine for now.

PlasmaPy gives each their own sub-packages, `plasmapy.analysis` and
`plasmapy.diagnostics` respectively.

Think of the `plasmapy.analysis` as your toolbox.  It has all the tools
(functionality) you need to analyze your data.  Functionality is built around

> `numpy` arrays and each function has a well-defined, focused task.

Reviewer: A possibility: link to `numpy.ndarray` instead of NumPy

Author: I made this a link to `numpy`'s page about their [array objects](https://numpy.org/doc/stable/reference/arrays.html).

> Diagnostics on the other-hand have a much broader scope and leverage the tools

Reviewer: ```suggestion
Diagnostics have a much broader scope and leverage the tools
```

defined in `plasmapy.analysis` to give a more integrated user experience when
analyzing data.  Diagnostics try to enhance the analysis workflow by focusing
on some of following key areas...

#. A more human-friendly way of managing data by building an interface around
   `xarray` arrays and datasets via custom diagnostic accessors.

   - `xarray` provides labelled multi-dimensional arrays and datasets.
   - Diagnostics self-manage the computed analysis data within a `xarray`
     dataset while maintaining the computed data's relation to the original
     data.

#. Quick viewing of analyzed data with default plotting routines.
#. Fully defining the physical parameters of a diagnostic with purposely

>    designed Probe classes that are integrated into the analysis workflow.

Reviewer: ```suggestion
   designed `Probe` classes that are integrated into the analysis workflow.
```
Link to the `Probe` classes directly?

> #. Adding GUI's to the analysis workflow via notebook widgets.

Reviewer: ```suggestion
#. Adding graphical user interfaces (GUIs) to the analysis workflow via notebook widgets.
```
I'm wondering if there could be newcomers to plasma science who are unfamiliar with the acronym.
